src/modules/search/hooks/useSearch.js

```diff
@@ -1,65 +1,48 @@
-/* eslint-disable complexity, max-statements, import/prefer-default-export */
-import { useMemo } from "react";
-import { API_VERSION } from "src/const/config";
-import { regex } from "src/const/regex";
-import { useCustomQuery } from "src/modules/common/hooks"
-import { validateAddress } from "src/utils/validators";
+import { regex } from 'src/const/regex';
+import { validateAddress } from 'src/utils/validators';
+import { useTransactions } from '@transaction/hooks/queries';
+import { useDelegates } from '@dpos/validator/hooks/queries';
+import { useBlocks } from '@block/hooks/queries/useBlocks';
 
+// eslint-disable-next-line complexity,max-statements
 export const useSearch = (search = '') => {
-  const config = {
-    method: 'get',
+  const isAddress = validateAddress(search) === 0;
+  const isTxId = regex.transactionId.test(search);
+  const isBlockHeight = regex.blockHeight.test(search);
+  const isDelegate = search.length >= 3 && !isAddress && !isTxId && !isBlockHeight;
+
+  const addresses = useDelegates({
+    config: { params: { address: search } },
+    options: { enabled: isAddress },
+  });
+
+  const delegates = useDelegates({
+    config: { params: { search } },
+    options: { enabled: isDelegate },
+  });
+
+  const transactions = useTransactions({
+    config: { params: { transactionID: search } },
+    options: { enabled: isTxId },
+  });
+
+  const blocks = useBlocks({
+    config: { params: { height: search } },
+    options: { enabled: isBlockHeight },
+  });
+
+  const isLoading =
+    delegates.isLoading || transactions.isLoading || addresses.isLoading || blocks.isLoading;
+
+  const isFetched =
+    delegates.isFetched && transactions.isFetched && addresses.isFetched && blocks.isFetched;
```

Comment:
@soroushm This was always be false because all the searches can't run at the same time. So I think it should be removed

Reply:
Isnt this more related to this https://github.com/LiskHQ/lisk-desktop/issues/4617 . Either way, we could fix it in that issue